velox/expression/tests/EvalSimplifiedTest.cpp

```diff
@@ -167,3 +167,51 @@ TEST_F(EvalSimplifiedTest, queryParameter) {
   auto* ptr = dynamic_cast<exec::ExprSetSimplified*>(exprSet.get());
   EXPECT_TRUE(ptr != nullptr) << "expected ExprSetSimplified derived object.";
 }
+
+TEST_F(EvalSimplifiedTest, specialFormPropagateNulls) {
```

Comment:
Any particular reason to add the test here and not in ExprTest? I believe there are no encodings in simplified eval.

Reply:
I mentioned it at the end of this comment at the top of the test, evalSimplified is especially prone to hitting this issue (and the test as I wrote it only hits the issue when run with evalSimplified) because it removes nulls so early in the evaluation.  It was much more challenging to write a test that would work in ExprTest (with eval).  I.e. I gave up and just used evalSimplified.